data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +756,246 @@ class ApiServerClient {
+    return this.getResource(_.merge(opts, {
+        namespaceId: namespaceId
+      }))
+      .then(resource => {
+        if (opts.spec && resource.spec) {
+          const spec = _.merge(resource.spec, camelcaseKeys(opts.spec));
+          _.set(opts, 'spec', spec);
+        }
+        return this.updateOSBResource(opts);
+      });
+  }
+
+  getAllServices() {
+    return Promise.try(() => this.init())
+      .then(() => apiserver.apis[CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.DEFAULT_NAMESPACE)[CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES].get())
+      .then(serviceList => _.get(serviceList, 'body.items'))
+      .then(serviceList => {
+        let services = [];
+        _.forEach(serviceList, service => {
+          services = _.concat(services, [service.spec]);
+        });
+        return services;
+      })
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      });
+  }
+
+  getAllPlansForService(serviceId) {
+    return Promise.try(() => this.init())
+      .then(() => apiserver.apis[CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.DEFAULT_NAMESPACE)[CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS].get({
+          qs: {
+            labelSelector: `serviceId=${serviceId}`
+          }
+        }))
+      .then(planList => planList.body.items)
+      .then(planList => {
+        let plans = [];
+        _.forEach(planList, plan => {
+          plans = _.concat(plans, [plan.spec]);
+        });
+        return plans;
+      })
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      });
+  }
+
+  /**
+   * @description Remove finalizers from finalizer list
+   * @param {string} opts.resourceGroup - Name of resource group 
+   * @param {string} opts.resourceType - Type of resource 
+   * @param {string} opts.resourceId - Unique id of resource
+   * @param {string} opts.finalizer - Name of finalizer
+   */
+  removeFinalizers(opts) {
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to remove finalizer`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to remove finalizer`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to remove finalizer`);
+    assert.ok(opts.finalizer, `Property 'finalizer' is required to remove finalizer`);
+    opts.namespaceId = opts.namespaceId ? opts.namespaceId : CONST.APISERVER.DEFAULT_NAMESPACE;
+    return this.getResource(opts)
+      .then(resourceBody => {
+        opts.metadata = {
+          resourceVersion: _.get(resourceBody, 'metadata.resourceVersion'),
+          finalizers: _.pull(_.get(resourceBody, 'metadata.finalizers'), opts.finalizer)
+        };
+        return this.updateOSBResource(opts);
+      });
+
+  }
+
+
+  /**
+   * @description Create Service/Plan Resource in Apiserver with given crd
+   */
+  createOrUpdateServicePlan(crd) {
```

Comment:
same comment as above. Should move to a higher construct. THis way, API Server Client will grow too long with every new CRD addition.

Reply:
@amitmalav can u check this.

Reply:
We don't have method already defined for simply making `post` or `patch` call.
```js
.then(() => apiserver
        .apis[CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR][CONST.APISERVER.API_VERSION]
        .namespaces(CONST.APISERVER.DEFAULT_NAMESPACE)[resourceType].post({
          body: crd
        }))
```
I feel it's okay if we keep this method here. If you still feel otherwise what we can do is
Have separate methods for each `put` `post` `patch`  and use them in all the methods. @subhankarc  WDYS?